staging/src/k8s.io/apiserver/pkg/server/options/tracing.go

```diff
@@ -123,9 +126,34 @@ func (o *TracingOptions) ApplyTo(es *egressselector.EgressSelector, c *server.Co
 	if c.LoopbackClientConfig != nil {
 		c.LoopbackClientConfig.Wrap(tracing.WrapperFor(c.TracerProvider))
 	}
+
+	// Also enable logging of span and trace ID.
+	//
+	// This has to run after applying the logging configuration, something
+	// that intentionally is done as early as possible before applying
+	// other configuration options.
+	logger := klog.Background()
+	logger = logger.WithContextHandlers(logr.ContextHandler{ValuesFromContext: traceIDsFromContext})
+	klog.SetLoggerWithOptions(logger, klog.ContextualLogger(true))
+
 	return nil
 }
 
+func traceIDsFromContext(ctx context.Context) logr.KeysAndValues {
+	spanContext := trace.SpanContextFromContext(ctx)
+	if !spanContext.IsValid() {
+		return nil
+	}
+
+	// A valid span context has both IDs set. These key names are
+	// determined by
+	// https://github.com/open-telemetry/opentelemetry-specification/tree/a87fb8c5a60f942d08c67c1582034eb8b4322009/specification/logs#trace-context-in-legacy-formats
+	return logr.KeysAndValues{
```

Comment:
would we want to return this as `"span", struct { trace_id, span_id string}{...}` or something, so they render as a struct?

Reply:
No, @dashpole wanted this to follow the OpenTelemetry spec, which implies that the IDs need to be at the top level, not embedded in some struct.